src/java/htsjdk/samtools/SRAIndex.java

```diff
@@ -0,0 +1,215 @@
+        long refEndPos = endPos; // TODO: check if that is a ref-local or global position  - refOffset;
+        if (refEndPos >= refLength) {
+            throw new RuntimeException("refEndPos is larger than reference length");
+        }
+
+        int firstBinNumber = (int)refStartPos / SRA_BIN_SIZE;
+        int lastBinNumber = (int)refEndPos / SRA_BIN_SIZE;
+
+        int numberOfBins = ((int)refLength / SRA_BIN_SIZE) + 1;
+
+        BitSet binBitSet = new BitSet();
+        binBitSet.set(0, SRA_BIN_INDEX_OFFSET, false);
+        if (firstBinNumber > 0) {
+            binBitSet.set(SRA_BIN_INDEX_OFFSET, SRA_BIN_INDEX_OFFSET + firstBinNumber, false);
+        }
+        binBitSet.set(SRA_BIN_INDEX_OFFSET + firstBinNumber, SRA_BIN_INDEX_OFFSET + lastBinNumber + 1, true);
+        if (lastBinNumber + 1 < numberOfBins) {
+            binBitSet.set(SRA_BIN_INDEX_OFFSET + lastBinNumber + 1, SRA_BIN_INDEX_OFFSET + numberOfBins, false);
+        }
+
+        return new BinList(referenceIndex, binBitSet);
+    }
+
+    @Override
+    public BAMFileSpan getSpanOverlapping(Bin bin) {
+        return new BAMFileSpan(getBinChunks(bin));
+    }
+
+    @Override
+    public BAMFileSpan getSpanOverlapping(int referenceIndex, int startPos, int endPos) {
+        BinList binList = getBinsOverlapping(referenceIndex, startPos, endPos);
+        BAMFileSpan result = new BAMFileSpan();
+        for (Bin bin : binList) {
+            result.add(getSpanOverlapping(bin));
+        }
+
+        return result;
+    }
+
+    @Override
+    public long getStartOfLastLinearBin() {
+        int numberOfReferences = recordRangleInfo.getReferenceLengths().size();
+        long refOffset = recordRangleInfo.getReferenceOffsets().get(numberOfReferences - 1);
+        long lastChunkNumber = recordRangleInfo.getReferenceLengths().get(numberOfReferences - 1) / SRA_CHUNK_SIZE;
+        return lastChunkNumber * SRA_CHUNK_SIZE + refOffset;
+    }
+
+    @Override
+    public BAMIndexMetaData getMetaData(int reference) {
+        throw new RuntimeException("Getting of BAM index metadata for SRA is not implemented");
+    }
+
+    @Override
+    public void close() { }
+
+    public static int getNumIndexLevels() {
+        return LEVEL_STARTS.length;
+    }
+
+    private List<Chunk> getBinChunks(Bin bin) {
+        if (bin.containsChunks()) {
+            return bin.getChunkList();
+        }
+
+        if (bin.getBinNumber() < SRA_BIN_INDEX_OFFSET) {
+            throw new RuntimeException("SRA only supports bins from the last level");
+        }
+        int binNumber = bin.getBinNumber() - SRA_BIN_INDEX_OFFSET;
+        long refOffset = recordRangleInfo.getReferenceOffsets().get(bin.getReferenceSequence());
+
+        // move requested position 5000 bases behind, so that we take all the reads that overlap requested position
+        int firstChunkCorrection = binNumber == 0 ? 0 : -5000;
```

Comment:
Extract this 5000 into a named constant at the top of the class, since it reflects assumptions about max read length.